Microsoft.Toolkit.Uwp/Helpers/ObjectStorage/BaseObjectStorageHelper.cs

```diff
@@ -73,13 +75,22 @@ public bool KeyExists(string compositeKey, string key)
         /// <returns>The T object</returns>
         public T Read<T>(string key, T @default = default(T))
         {
-            string value = (string)Settings.Values[key];
-            if (value != null)
+            object value = null;
+
+            if (!Settings.Values.TryGetValue(key, out value))
             {
-                return JsonConvert.DeserializeObject<T>(value);
+                return @default;
             }
 
-            return @default;
+            var type = typeof(T);
+            var typeInfo = type.GetTypeInfo();
+
+            if (typeInfo.IsPrimitive || type == typeof(string))
+            {
+                return (T)Convert.ChangeType(value, type);
```

Comment:
I suppose this will create breaking changes if there is already data saved with a previous version of the `Save` method. Right?

Reply:
that's what i am trying to avoid by executing ChangeType.. not to break existing thing. Sadly i don't use it. Previously all data saved as string. This works with converting string to say int. Didn't try other types though they fallback to original json serialiser way

Reply:
Could this throw an exception if the cast is not possible? If so, should it return @default in that case?

Reply:
https://msdn.microsoft.com/en-us/library/dtb69x08(v=vs.110).aspx?cs-save-lang=1&cs-lang=csharp#code-snippet-1

We are calling this only for primitive / string type. Convert supports these types.

I will have a look wrt null value.

Reply:
Yeah, I'm just wondering if **value** could be a complex object at this point (complex object could have been saved with that key) and if so will it throw an exception?.

Reply:
how about
```
if(value is string strVal)
{
   return JsonConvert.DeserializeObject<T>(strVal);
}
return (T)value;
```

Reply:
Could even improve the if for string types
```
if((type != typeof(string)) && (value is string strVal))
```

Reply:
But I would hope that Json.Net has a fast exit if the deserialized type is string

Reply:
Making some unit tests can do the trick.

Reply:
^ THIS!!!

Reply:
Good idea

Reply:
Being the kinda obtuse person I am, i have only added null check for value to ensure exception isn't thrown @nmetulev 
I have also added unit tests to test both legacy and new behaviour @skendrot @Odonno